### PR TITLE
Create an XLA parameter and fix the mixed precision

### PR DESCRIPTION
**Pull request by jplu**
This PR adds a new `XLA` parameter to activate/deactivate the XLA compilation and a bug in the mixed precision. These have to be set before the creation of the strategy and float16 is not compliant with TPU, where only bfloat16 is available.

### CODE REVIEW

File: src/transformers/training_args_tf.py [hunk 1]
```diff
@@ -110,12 +110,23 @@ class TFTrainingArguments(TrainingArguments):
         metadata={"help": "Name of TPU"},
     )
 
+    xla: bool = field(default=False, metadata={"help": "Whether to activate the XLA compilation or not"})
```
Comment (sgugger) [inline src/transformers/training_args_tf.py:113]: Can you document it in the docstring too?

**Review by sgugger — APPROVED** [review]
LGTM! Just one nit for the documentation.

**Review by LysandreJik — APPROVED** [review]
LGTM, very cool!

File: src/transformers/training_args_tf.py (hunk 1 above)
Comment (jplu) [reply src/transformers/training_args_tf.py:115]: Should be ok.

**Review by LysandreJik — APPROVED** [review]
Thanks!

**Pull request merged**